### PR TITLE
Adding subtitle to ExpansionTile

### PR DESCRIPTION
**Pull request by luizfranca**
## Description

As per request to make ExpansionTile more flexible this PR is adding subtitle to ExpansionTile.

## Related Issues

[#15427](https://github.com/flutter/flutter/issues/15427)

## Tests

I added the following tests:

- Added a subtitle to an existing test.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/flutter/lib/src/material/expansion_tile.dart [hunk 1]
```diff
@@ -53,6 +54,11 @@ class ExpansionTile extends StatefulWidget {
   /// Typically a [Text] widget.
   final Widget title;
 
+  /// Additional content displayed bellow the title
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/expansion_tile.dart]:
```suggestion
  /// Additional content displayed below the title.
```

File: packages/flutter/test/material/expansion_tile_test.dart [hunk 2]
```diff
@@ -70,6 +70,7 @@ void main() {
                 key: expandedKey,
                 initiallyExpanded: true,
                 title: const Text('Expanded'),
+                subtitle: const Text('Subtitle'),
```
Comment (shihaohong) [inline packages/flutter/test/material/expansion_tile_test.dart]: I don't think you should add this line here. We should assume that the default `ExpansionTile` does not have its subtitle defined. 

Instead, we should write a new test that validates that if ExpansionTile.subtitle is defined, it is indeed shown.

Comment (shihaohong) [reply packages/flutter/test/material/expansion_tile_test.dart]: It seems that this line is also breaking the presubmit tests.

File: packages/flutter/test/material/expansion_tile_test.dart [hunk 3]
```diff
@@ -215,4 +215,35 @@ void main() {
     expect(iconColor(expandedIconKey), _unselectedWidgetColor);
     expect(iconColor(collapsedIconKey), _accentColor);
   });
+
+  testWidgets('ExpansionTile subtitle', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      MaterialApp(
```
Comment (shihaohong) [inline packages/flutter/test/material/expansion_tile_test.dart]: It there a need for all the extra config? I would imagine that the following would suffice: 
```
      MaterialApp(
        home: Scaffold(
          body: ExpansionTile(
            title: const Text('Title'),
            subtitle: const Text('Subtitle'),
            children: const <Widget>[ListTile(title: Text('0'))],
          ),
        ),
      ),
```

This also makes your test's intent clear to future devs reading this.

Comment (luizfranca) [reply packages/flutter/test/material/expansion_tile_test.dart]: It isn't, initially I thought in doing this way, but I was unsure whether this additional config was wanted. I'm going to update it.

**Comment by shihaohong** [comment]
You seem to be failing a few analyzer tests. You can run the analyzer locally through `flutter analyze --flutter-repo`

**Review by shihaohong — APPROVED** [review]
LGTM

**Pull request merged**